project/mmi366127/README.md

```diff
@@ -0,0 +1,107 @@
+# Geometric search engine
+
+A project that implement R-tree, kd-tree, M-tree for geometric searching.
+
+## Basic Information
+GitHub repository: [Geometric search engine](https://github.com/mmi366127/Geometric-Search-Engine)
+
+Geometric search is important for implmenting map and geometry system. For more efficient query, there are many data structures for different circumstances, including R-tree, kd-tree and M-tree. Moreover, these data structures can also be used to do machine learning tasks in KNN(K-nearest-neighbors) algorithm. Therefore, in this project, I am going to implement an API for geometric search for arbitrary dimension data. 
+
+## Problem to Solve
+
+Using R-tree, kd-tree and M-tree to do query on nearest-n-points, and range-query. This project assumes using Euclidean distance for distance metric(could add more distance metric if possible.)
+
+## Prospective Users
+
+For users that aims to build their geometric information system, the API can provide useful tools to do the searching. And for machine learning users, the API can provide high dimension support for their usage.
+
+## System Architeure
+
+- Using ``C++`` to implement the algorithm.
+- Using ``pybind11`` to make API in python.
+
+
+## API Description
+
+- Using ``vector<double>`` for coordinate and ``int`` for ID to represent the data points in C++.
```

Comment:
Why do you choose the platform dependent data type for the index?

Reply:
Then, I think I should change it to ``int32_t``?